examples/webgpu_tsl_editor.html

```diff
@@ -0,0 +1,218 @@
+
+				const width = 200;
+				const height = 200;
+
+				const camera = new THREE.PerspectiveCamera( 70, width / height, 0.1, 10 );
+				camera.position.z = .72;
+				camera.lookAt( 0, 0, 0 );
+
+				const scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0x222222 );
+
+				const rendererDOM = document.getElementById( 'renderer' );
+
+				const renderer = new WebGPURenderer();
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( 200, 200 );
+				rendererDOM.appendChild( renderer.domElement );
+
+				const material = new NODES.NodeMaterial();
+				material.colorNode = NODES.vec4( 0, 0, 0, 1 );
+
+				const mesh = new THREE.Mesh( new THREE.PlaneGeometry( 1, 1 ), material );
+				scene.add( mesh );
+
+				renderer.setAnimationLoop( () => {
+
+					renderer.render( scene, camera );
+
+				} );
+				// editor
+
+				window.require.config( { paths: { 'vs': 'https://cdn.jsdelivr.net/npm/monaco-editor@latest/min/vs' } } );
+
+				require( [ 'vs/editor/editor.main' ], () => {
+
+					const options = { shader: 'fragment' };
+
+					let timeout = null;
+
+					const editorDOM = document.getElementById( 'source' );
+					const resultDOM = document.getElementById( 'result' );
+
+					const tslCode = `// Example: Desaturate texture
+
+const { texture, uniform, vec4 } = TSL;
+
+//const samplerTexture = new THREE.Texture();
+const samplerTexture = new THREE.TextureLoader().load( './textures/uv_grid_opengl.jpg' );
+
+// label is optional
+const myMap = texture( samplerTexture ).rgb.label( 'myTexture' );
+const myColor = uniform( new THREE.Color( 0x0066ff ) ).label( 'myColor' );
+const opacity = .7;
+
+const desaturatedMap = myMap.rgb.saturation( 0 ); // try add .temp( 'myVar' ) after saturation()
+
+const finalColor = desaturatedMap.add( myColor );
+
+output = vec4( finalColor, opacity );
+`;
+
+					const editor = window.monaco.editor.create( editorDOM, {
+						value: tslCode,
+						language: 'javascript',
+						theme: 'vs-dark',
+						automaticLayout: true
+					} );
+
+					const result = window.monaco.editor.create( resultDOM, {
+						value: '',
+						language: 'wgsl',
+						theme: 'vs-dark',
+						automaticLayout: true,
+						readOnly: true
+					} );
+
+					const build = () => {
+
+						try {
+
+							const tslCode = `let output = null;\n${ editor.getValue() }\nreturn { output };`;
+							const nodes = new Function( 'THREE', 'TSL', tslCode )( THREE, NODES, tslCode );
+
+							mesh.material.colorNode = nodes.output;
+							mesh.material.dispose();
+
+							const outputColorSpace = renderer.outputColorSpace;
+							renderer.outputColorSpace = THREE.NoColorSpace; // Don't apply "sRGB/Display-P3" encoding to clean up the output.
+
+							const nodeBuilder = new WGSLNodeBuilder( mesh, renderer );
+							nodeBuilder.build();
+
+							result.setValue( nodeBuilder[ options.shader + 'Shader' ] );
+							result.revealLine( 1 );
+
+							renderer.outputColorSpace = outputColorSpace; // restore
```

Comment:
Maybe we should leave it at `NoColorSpace`? Because if user supplies a shader that doesn't use sRGB conversion then they doesn't expect it will be implicitly converted, and if the shader does they should be fine.

Reply:
I will create an option to toggle.

Reply:
I will create an option to toggle.